src/renderer/widgets/PasswordInputField.cpp

```diff
@@ -304,6 +305,11 @@ bool CPasswordInputField::draw(const SRenderData& data) {
     return dots.currentAmount != passwordLength || fade.animated || col.animated || redrawShadow || data.opacity < 1.0 || forceReload;
 }
 
+static void failTimeoutCallback(std::shared_ptr<CTimer> self, void* data) {
+    g_pAuth->m_bDisplayFailText = false;
+    g_pHyprlock->renderAllOutputs();
+}
+
```

Comment:
So the thing about moving this to `PasswordInputField` is that it will be called for each monitor separately. So you would have one timer per monitor. Not that big of a deal, but i think it is better to just have it in the `onPasswordCheckTimer` function to avoid that.

Reply:
Tbh it has been moved just to be consistent with the configuration parameters which is input-field->fail_timeout.
Not sure what's the proper way to access that option outside `CPasswordInputField`.

Reply:
Yeah makes sense, since you could have multiple input fields theoretically. I think it is fine link this.